src/Matrix.php

```diff
@@ -957,6 +957,110 @@ public function matmul(Matrix $b) : self
         return self::quick($c);
     }
 
+    /**
+     * Multiply this matrix with another matrix using the Strassen algorithm (matrix-matrix product).
+     *
+     * @param \Tensor\Matrix $b
+     * @throws \InvalidArgumentException
+     * @return self
+     */
+    public function matmulStrassen(Matrix $b) : self
+    {
+        if ($this->n !== $b->m()) {
+            throw new InvalidArgumentException('Matrix A requires'
+                . " $this->n rows but Matrix B has {$b->m}.");
+        }
+
+        if ($this->n === 1) {
+            return self::quick([[$this->a[0][0] * $b->a[0][0]]]);
+        }
+
+        if ($this->n % 2 !== 0) {
+            $a1 = $this->a;
+            $b1 = $b->a;
+
+            for ($i = 0; $i < $this->n + 1; ++$i) {
+                for ($j = 0; $j < $this->n + 1; ++$j) {
+                    $a1[$i][$j] = $a1[$i][$j] ?? 0;
+                    $b1[$i][$j] = $b1[$i][$j] ?? 0;
+                }
+            }
+
+            $aM = self::quick($a1);
+            $bM = self::quick($b1);
+
+            $c = $aM->matmulStrassen($bM);
+
+            $c1 = [];
+            for ($i = 0; $i < $this->n; ++$i) {
+                for ($j = 0; $j < $this->n; ++$j) {
+                    $c1[$i][$j] = $c->a[$i][$j];
+                }
+            }
+
+            return self::quick($c1);
+        }
+
+        /** @var int $size */
+        $size = $this->n / 2;
+        $a11 = $a12 = $a21 = $a22 = $b11 = $b12 = $b21 = $b22 = [];
+
+        for ($i = 0; $i < $size; ++$i) {
+            for ($j = 0; $j < $size; ++$j) {
+                $a11[$i][$j] = $this->a[$i][$j];
+                $a12[$i][$j] = $this->a[$i][$j + $size];
+                $a21[$i][$j] = $this->a[$i + $size][$j];
```

Comment:
Looks like some of these simple operations can be precomputed for better performance (it's not obvious that the PHP runtime will optimize these for us).

Ex.

```php
        for ($i = 0; $i < $size; ++$i) {
            $iSize = $i + $size; // Precompute value in the outer loop, instead of inner
           
            for ($j = 0; $j < $size; ++$j) {
                $a11[$i][$j] = $this->a[$i][$j];
                $a12[$i][$j] = $this->a[$i][$j + $size];
                $a21[$i][$j] = $this->a[$iSize][$j];
```